File: 2023/2023W02/preppindataW02.R
library(dplyr)
library(readr)

final <- read_csv("C:/Data/PreppinData/Transactions.csv") %>%
  merge(read_csv("C:/Data/PreppinData/Swift Codes.csv"), by='Bank') %>%
  mutate('Sort Code' = gsub('-','',`Sort Code`),
         'IBAN' = paste0('GB', `Check Digits`, `SWIFT code`, `Sort Code`, `Account Number`)) %>%
  select(c('Transaction ID', 'IBAN'))
  
View(final)

File: 2023/2023W02/PreppindataW02.py
import pandas as pd

transaction = pd.read_csv(r'C:/Data/PreppinData/Transactions.csv')
swift = pd.read_csv(r'C:/Data/PreppinData/Swift Codes.csv')

country_code = {'UK':'GB'}
final = pd.merge(transaction, swift, on='Bank')
final['IBAN'] = country_code['UK'] + final['Check Digits'] + final['SWIFT code'] \
    + final['Sort Code'].str.replace('-','') + final['Account Number'].astype(str)
final = final[['Transaction ID', 'IBAN']].copy()
